### PR TITLE
PathDriver interface to manipulate system paths

### PR DESCRIPTION
**Pull request by gupta-ak**
Removes implementation from #79 .  Adds a path driver interface and a basic implementation similar to driver.go.

### CODE REVIEW

**Comment by gupta-ak** [comment]
@stevvooe I added the package changes back in.

File: conterrors/error.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package conterrors
```
Comment (stevvooe) [inline conterrors/error.go]: Don't make a package for this: just define the errors where they are thrown and keep a helper function that handles the various types.

File: pathdriver/path_driver.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	IsAbs(path string) bool
+	Rel(base, target string) (string, error)
+	Base(path string) string
+	Dir(path string) string
+	Clean(path string) string
+	Split(path string) (dir, file string)
+	Separator() byte
+	Abs(path string) (string, error)
+	Walk(string, filepath.WalkFunc) error
+	NormalizePath(path string) string
```
Comment (stevvooe) [inline pathdriver/path_driver.go]: Since this isn't a direct abstraction to filepath, make sure to document the expectations, or just call it `FromSlash`.

**Comment by stevvooe** [comment]
LGTM

@dmcgowan PTAL

**Comment by AkihiroSuda** [comment]
ping @dmcgowan

**Comment by dmcgowan** [comment]
LGTM

**Pull request merged**